src/platform/OpenThread/GenericNetworkCommissioningThreadDriver.cpp

```diff
@@ -119,9 +119,9 @@ Status GenericThreadDriver::AddOrUpdateNetwork(ByteSpan operationalDataset, Muta
     // Staging network not commissioned yet (active) or we are updating the dataset with same Extended Pan ID.
     VerifyOrReturnError(!mStagingNetwork.IsCommissioned() || MatchesNetworkId(mStagingNetwork, newExtpanid) == Status::kSuccess,
                         Status::kBoundsExceeded);
+    mStagingNetwork = newDataset;
```

Comment:
No, this was supposed to back up the previous configuration, not the one you're setting. Could you improve the PR description and tell exactly which scenario is failing for you?